app/models/concerns/hostext/search.rb

```diff
@@ -171,6 +171,20 @@ def search_by_config_group(key, operator, value)
         {:conditions => opts, :include => :hostgroup}
       end
 
+      def cast_facts(key, operator, value)
+        begin
+          is_int = !!Integer(value)
```

Comment:
> you're right, it's slightly faster but in this case we don't do the operation for 10_000 records but for 1 which IMHO does not justify using exceptions for driving program flow

@ares I usually try to avoid un-necessary regex, but I don't mind changing to regex if you want. I will change this later. I am now more concerned on how to approach the PG issue :)